### DISCUSSION TITLE
Pulsar in Active , Active small scale Setup

### DISCUSSION THREAD

**Discussion by nareshv** [apache/pulsar#22581] [Q&A]
So far we have had good success running pulsar standalone on one VM. We are looking for ways to run 2 {Brokers, Bookies} with Synchronous replication between them and still sustain 1 VM Loss.

Any guidance on this ?

**Comment by lhotari** [comment]
Assuming that this is a local redundancy setup and not about geographical redundancy where Pulsar's geo-replication would be relevant.

Pulsar standalone shouldn't be considered a unit of deployment. As mentioned before, Pulsar standalone is not designed to be used for production use, but if it works for you, we won't stop you.

Instead of thinking of redundancy with 2 Pulsar Standalone instances and having replication between them, a better approach would be to run a minimal setup where there's redundancy using the standard features of Bookkeeper and Zookeeper. For bookies, it's fine to have 2 instances for redundancy, but for Zookeeper you would need 3 instances to reach quorum in a redundant setup.
